src/gf.c

```diff
@@ -596,8 +596,8 @@ static void jl_cacheable_sig(
         // avoid specializing on an argument of type Tuple
         // unless matching a declared type of `::Type`
         if (jl_is_type_type(elt) && jl_is_tuple_type(jl_tparam0(elt)) &&
-            (!jl_subtype(decl_i, (jl_value_t*)jl_type_type) || is_kind(decl_i))) { // Type{Tuple{...}}
-            elt = (jl_value_t*)jl_anytuple_type_type; // Type{T<:Tuple}
+            (!jl_subtype(decl_i, (jl_value_t*)jl_type_type) || jl_is_kind(decl_i))) { // Type{Tuple{...}}
+            elt = (jl_value_t*)jl_anytuple_type_type; // Type{T} where T<:Tuple
```

Comment:
github may have lost my previous comment here under the fold when you changed the comment on this line maybe you already have this change locally?):
> since jl_anytuple_type_type is no longer a singleton, this test needs to be updated to test for type equality between elt and Type{T} where T<:Tuple